core/src/avm2/object/error_object.rs

```diff
@@ -0,0 +1,121 @@
+use crate::avm2::activation::Activation;
+use crate::avm2::call_stack::StackTrace;
+use crate::avm2::object::script_object::ScriptObjectData;
+use crate::avm2::object::{ClassObject, Object, ObjectPtr, TObject};
+use crate::avm2::value::Value;
+use crate::avm2::Error;
+use crate::string::{AvmString, WString};
+use gc_arena::{Collect, GcCell, MutationContext};
+use std::cell::{Ref, RefMut};
+
+/// A class instance allocator that allocates Error objects.
+pub fn error_allocator<'gc>(
+    class: ClassObject<'gc>,
+    proto: Object<'gc>,
+    activation: &mut Activation<'_, 'gc, '_>,
+) -> Result<Object<'gc>, Error> {
+    let base = ScriptObjectData::base_new(Some(proto), Some(class));
+
+    Ok(ErrorObject(GcCell::allocate(
+        activation.context.gc_context,
+        ErrorObjectData {
+            base,
+            id: 0,
+            message: AvmString::default(),
+            name: "Error".into(),
+            stack_trace: activation.id.to_stack_trace(),
+        },
+    ))
+    .into())
+}
+#[derive(Clone, Collect, Debug, Copy)]
+#[collect(no_drop)]
+pub struct ErrorObject<'gc>(GcCell<'gc, ErrorObjectData<'gc>>);
+
+#[derive(Clone, Collect, Debug)]
+#[collect(no_drop)]
+pub struct ErrorObjectData<'gc> {
+    /// Base script object
+    base: ScriptObjectData<'gc>,
+    name: AvmString<'gc>,
+    message: AvmString<'gc>,
+    id: i32,
+    stack_trace: StackTrace<'gc>,
+}
```

Comment:
The only native data in the Error object is the stack trace. The rest of the properties like `name` and `id` are normal dumb properties on the object as if it were any other Object, not built-ins:

```
    public var message;
      
      public var name;
      private var _errorID:int;
      
      public function get errorID() : int
      {
         return this._errorID;
      }
```
So they don't need to be in `ErrorObject`.